### DISCUSSION TITLE
enforce equidistant placement

### DISCUSSION THREAD

**Discussion by salmen20** [The-OpenROAD-Project/OpenROAD-flow-scripts#2576] [General] [answered]
Hello everyone,
I'm working on a design in OpenROAD and I would like to place certain cells at equal distances from a reference cell. Can anyone guide me on how to enforce equidistant placement for cells relative to another reference cell within the OpenROAD flow? Are there specific constraints or placement strategies I can use to achieve this?
Thanks.

**Comment by maliberty** [comment]
There is no such constraint.  Can you pre-place these instances?

**Reply by salmen20** [reply]
How can i do that ? you mean placing them manually ?
Thanks

**Reply by maliberty** [reply, marked as answer]
Either through a floorplan DEF where they are placed and marked FIXED or through tcl, eg

```
proc place_inst { inst x y } {
  set db_inst [[ord::get_db_block] findInst $inst]
  $db_inst setLocation $x $y
  $db_inst setOrient R0
  $db_inst setPlacementStatus LOCKED
}
```

**Reply by salmen20** [reply]
I found an example with the chameleon design i guess it can solve my probem but will take a lot of time as i have a lot of instance , i will try to find a way to automate  that , the question here in order to get those name will be by running the synth getting those names , then running floorplan with the MACRO_PLACEMENT variable ?

**Reply by maliberty** [reply]
You would need the post-synth names.  If these are macros then you could use MACRO_PLACEMENT

**Reply by salmen20** [reply]
I will try that and see , Thank you

**Reply by salmen20** [reply]
> You would need the post-synth names. If these are macros then you could use MACRO_PLACEMENT

i tried that and its working but it will take a lot of time to do it , is there a way i can specify a region for a certain modules ? i can get my top module using the getModInsts but i didnt find a way to place them as whole module in a certain region without defining them as a macro and thanks
